twisted/test/proto_helpers.py

```diff
@@ -695,3 +702,86 @@ def connectUNIX(self, address, factory, timeout=30, checkPID=0):
         Fake L{IReactorUNIX.connectUNIX}, that raises L{_connectException}.
         """
         raise self._connectException
+
+
+
+@implementer(ILogFilterPredicate)
+class NamespaceFilterPredicate(object):
+    """
+    L{ILogFilterPredicate} that filters out events not matching configured
+    namespace.
+    """
+
+    def __init__(self, namespace):
+        self.namespace = namespace
+
+
+    def __call__(self, event):
+        namespace = event.get("log_namespace", None)
+
+        if namespace != self.namespace:
+            return PredicateResult.no
+
+        return PredicateResult.maybe
+
+
+
+class LogCapture(object):
+    """
+    A context manager that captures log events and makes them available in
+    plain text format or as raw events list.
+
+    Can be used like this:
+
+    with LogCapture() as lc:
+        [... code that logs events ...]
+
+        events = lc.events
+        messages = lc.messages()
+        self.assertIn('logline text', messages[0])
+        self.assertNotIn('Unable to format event', messages[0])
+        selc.assertEqual(events['log_level'], Loglevel.debug)
+
+    """
+
+    def __init__(self, namespace=None):
+        self.namespace = namespace
+        self._events = []
+
+        self._observer = lambda e: self._events.append(e)
+
+        if namespace is not None:
+            self._observer = FilteringLogObserver(
+                self._observer, [NamespaceFilterPredicate(namespace)]
+            )
+
+
+    def __enter__(self):
+        globalLogPublisher.addObserver(self._observer)
+        return self
+
+
+    def __exit__(self, type_, value_, tb_):
+        globalLogPublisher.removeObserver(self._observer)
+
+
+    @property
+    def events(self):
+        """
+        Get captured events as a list of event dicts.
+
+        @return: The list of captured events.
+        @rtype: L{list} of L{dict}
+        """
+        return self._events[:]
+
+
+    @property
+    def messages(self):
```

Comment:
Just a suggestion here - do you think it might be clearer to call this `eventsAsText`?  "messages" could just be a synonym for "events", vernacularly.